### PR TITLE
changes in verdi restapi command (#2384)

### PR DESCRIPTION
**Pull request by waychal**
- updated verdi restapi command and its click parameters
- removed load_profile functionality from restapi code as it is already handled by verdi command
- removed \_\_main\_\_ from restapi code as it is duplicating the functionality provided by 'verdi restapi' command

### CODE REVIEW

**Comment by waychal** [comment]
@sphuber I have made the requested changes.

File: aiida/cmdline/commands/cmd_restapi.py [hunk 1]
```diff
@@ -26,34 +26,44 @@
 
 
 @verdi.command('restapi')
-@click.option('-H', '--host', type=click.STRING, default='127.0.0.1', help='the hostname to use')
-@click.option('-p', '--port', type=click.INT, default=5000, help='the port to use')
+@click.option('-H', '--hostname', type=click.STRING, default='127.0.0.1', help='Hostname')
```
Comment (sphuber) [inline aiida/cmdline/commands/cmd_restapi.py]: please use the predefined `aiida.cmdline.params.options.HOSTNAME` and `aiida.cmdline.params.options.PORT` for these. Note you can override the default when you call it. See other verdi commands for examples

File: aiida/cmdline/commands/cmd_restapi.py [hunk 2]
```diff
@@ -26,34 +26,44 @@
...
     from aiida.restapi.run_api import run_api
 
     # Construct parameter dictionary
     kwargs = dict(
-        hookup=True,
         prog_name='verdi-restapi',
-        default_host=host,
-        default_port=port,
+        hostname=hostname,
+        port=port,
```
Comment (sphuber) [inline aiida/cmdline/commands/cmd_restapi.py]: the same goes for the `default_config` no? As in, this should just be `config`

File: aiida/restapi/run_api.py [hunk 3]
```diff
@@ -16,127 +16,47 @@
...
-
-    if parse_aiida_profile and parsed_args.aiida_profile is not None:
-        aiida_profile = parsed_args.aiida_profile
-
-    elif confs.DEFAULT_AIIDA_PROFILE is not None:
-        aiida_profile = confs.DEFAULT_AIIDA_PROFILE
+    # Unpack parameters
+    hostname = kwargs['hostname']
+    port = kwargs['port']
+    default_config_dir = kwargs['default_config']
```
Comment (sphuber) [inline aiida/restapi/run_api.py]: Same here

File: aiida/restapi/run_api.py [hunk 4]
```diff
@@ -45,98 +44,20 @@ def run_api(flask_app, flask_api, *args, **kwargs):
...
-        aiida_profile = confs.DEFAULT_AIIDA_PROFILE
+    # Unpack parameters
+    default_host = kwargs['default_host']
+    default_port = kwargs['default_port']
+    default_config_dir = kwargs['default_config']
 
-    else:
-        aiida_profile = "default"
+    catch_internal_server = kwargs['catch_internal_server']
+    debug = kwargs["debug"]
```
[the remark this answers is not in the document]
Comment (waychal) [reply aiida/restapi/run_api.py]: Done

File: aiida/cmdline/commands/cmd_restapi.py [hunk 5]
```diff
@@ -26,33 +26,44 @@
...
     help='the path of the configuration directory')
-def restapi(host, port, config_dir):
+@click.option('-d', '--debug', type=click.BOOL, default=False, help='run app in debug mode')
+@click.option(
+    '-w',
+    '--wsgi-profile',
+    type=click.BOOL,
+    default=False,
+    help='to use WSGI profiler middleware for finding bottlenecks in web application')
+@click.option('--hookup', type=click.BOOL, default=True, help='to hookup app')
```
[the remark this answers is not in the document]
Comment (waychal) [reply aiida/cmdline/commands/cmd_restapi.py]: Done

File: aiida/cmdline/commands/cmd_restapi.py [hunk 6]
```diff
@@ -26,33 +26,44 @@
...
+def restapi(host, port, config_dir, debug, wsgi_profile, hookup):
     """
     Run the AiiDA REST API server
 
     Example Usage:
 
         \b
         verdi -p <profile_name> restapi --host 127.0.0.5 --port 6789 --config-dir <location of the config.py file>
+        --debug True --wsgi-profile False --hookup True
```
[the remark this answers is not in the document]
Comment (waychal) [reply aiida/cmdline/commands/cmd_restapi.py]: Done

File: aiida/restapi/run_api.py [hunk 7]
```diff
@@ -45,98 +44,20 @@ def run_api(flask_app, flask_api, *args, **kwargs):
...
-    elif confs.DEFAULT_AIIDA_PROFILE is not None:
-        aiida_profile = confs.DEFAULT_AIIDA_PROFILE
+    # Unpack parameters
+    default_host = kwargs['default_host']
+    default_port = kwargs['default_port']
+    default_config_dir = kwargs['default_config']
 
-    else:
-        aiida_profile = "default"
+    catch_internal_server = kwargs['catch_internal_server']
```
[the remark this answers is not in the document]
Comment (waychal) [reply aiida/restapi/run_api.py]: Done

File: aiida/restapi/run_api.py [hunk 8]
```diff
@@ -45,98 +44,20 @@ def run_api(flask_app, flask_api, *args, **kwargs):
...
-    #
-    # default restapi profile set in <config_dir>/config.py
-
-    if parse_aiida_profile and parsed_args.aiida_profile is not None:
-        aiida_profile = parsed_args.aiida_profile
-
-    elif confs.DEFAULT_AIIDA_PROFILE is not None:
-        aiida_profile = confs.DEFAULT_AIIDA_PROFILE
+    # Unpack parameters
+    default_host = kwargs['default_host']
```
[the remark this answers is not in the document]
Comment (waychal) [reply aiida/restapi/run_api.py]: Done

File: aiida/cmdline/commands/cmd_restapi.py [hunk 9]
```diff
@@ -26,33 +26,44 @@
...
     # Construct parameter dictionary
     kwargs = dict(
-        hookup=True,
         prog_name='verdi-restapi',
         default_host=host,
         default_port=port,
         default_config=config_dir,
+        debug=debug,
+        wsgi_profile=wsgi_profile,
+        hookup=hookup,
```
[the remark this answers is not in the document]
Comment (waychal) [reply aiida/cmdline/commands/cmd_restapi.py]: Done

File: aiida/cmdline/commands/cmd_restapi.py (hunk 2 above)
Comment (waychal) [reply aiida/cmdline/commands/cmd_restapi.py]: Corrected.

File: aiida/restapi/run_api.py (hunk 3 above)
Comment (waychal) [reply aiida/restapi/run_api.py]: Done

File: aiida/cmdline/commands/cmd_restapi.py (hunk 1 above)
Comment (waychal) [reply aiida/cmdline/commands/cmd_restapi.py]: I have made changes for this but I am not sure if it is correct. Please let me know if I am missing anything again.

Comment (sphuber) [reply aiida/cmdline/commands/cmd_restapi.py]: nope, all good :+1:

**Comment by sphuber** [comment]
Only your pre-commit is failing. Make sure you have updated your dependencies: `pip install -e .[dev_precommit] --upgrade`. If you fix those and push, I will approve. Thanks!

**Comment by sphuber** [comment]
Excellent! Thanks a lot @waychal

**Pull request merged**